fftools/ffmpeg.c

```diff
@@ -1291,6 +1290,8 @@ static void do_video_out(OutputFile *of,
 
         while (1) {
             ret = avcodec_receive_packet(enc, &pkt);
+            if (pkt.duration == AV_NOPTS_VALUE || pkt.duration <= 0)
+                pkt.duration = av_rescale_q(in_picture->pkt_duration, ost->mux_timebase, enc->time_base);
```

Comment:
[from IRL discussion] jagraff explained to me that this is probably not the 'right' way to copy the duration between frames (since it's probably better to figure out why the duration isn't copied over and fix that). So this is unlikely to be merged into mainline - but I think it's fine to use this for GIPHY's purposes.